submitting the assignment

EliSzBr opened the pull request:
issues I had -
overthinking things at the start - designing something overbuilt
due to that - slightly short on time which resulted in lack of optimizations:
- large image sizes due to sqlalchemy package requirement issues.
- os package problem - hence database connection details are not env variables but hardcoded.
- slow build time.
- I should have read into the connection pool etc.. in sqlalchemy engine instead of the easy  - but perhaps flawed - fix of removing the connection open/close from the sniffer loop for every packet.

amit7itz commented on server/main.py:
```diff
@@ -0,0 +1,40 @@
+
+@app.route("/all")
+def all():
+	session = create_session()
```
you can use `with create_session() as session:` so it will close on exit

amit7itz commented on server/main.py:
```diff
@@ -0,0 +1,40 @@
+
+@app.route("/routers")
+def routers():
+	ip_address_match = 3
```
consts should be upper case and at the top of the file (after imports)

amit7itz commented on sniffer/sniffer.py:
```diff
@@ -33,20 +34,23 @@ def process_data():
-    parser.add_argument("--clean", const=True, default=False, nargs='?', help="prune the existing data before starting")
+    parser.add_argument("--clean", const=True, default=False, nargs='?', help="prune the existing data before start")
     parser.add_argument("--debug", const=True, default=False, nargs='?', help="enable debug logging")
+    parser.add_argument("--clean_only", const=True, default=False, nargs='?', help="prune the existing data and exit")
```
lol that's confusing. If I only add `--clean_only` it doesn't clean :O

amit7itz reviewed (commented):
Nice work :)

EliSzBr commented:
all of the issues raised in the PR were fixed. 
also added logging for the webserver
fixed 500 due to pseudocode in the /lastseen route of the webserver (just returns not implemented now) - probably a better practice for automatic testing is actually to return 500/404? IDK..
